src/Paprika/Chain/Blockchain.cs

```diff
@@ -393,10 +393,12 @@ public void Finalize(Keccak keccak)
     /// </summary>
     private class BlockState : RefCountingDisposable, IWorldState, ICommit, IProvideDescription, IStateStats
     {
+        [ThreadStatic]
+        private static HashSet<ulong>? s_bloomCache;
```

Comment:
Have you measured it? It could save a lot of memory as usually a single BlockState will be out there.
At the same time, maybe some Resettable should be used. `HashSet` do not trim on `Clear`.

Reply:
Yep, finding changes with a profiler sitting on maintnet 😉

HashSet was always allocating a lot in AddWithResize; as it starts at 0 elements. However if started with a high capacity then it would always allocate a large amount, so this seems a fair comprimize.

This HashSet is in BlockProcessing, ConcurrentDictionary allocations were in TxPool